app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewState.kt

```diff
@@ -18,3 +19,8 @@ sealed class ConversationAvatar {
     class OneOne(val avatarAsset: UserAvatarAsset?) : ConversationAvatar()
     class Group(val groupColorValue: Long) : ConversationAvatar()
 }
+
+sealed class DownloadedAssetDialogVisibilityState {
+    object Hidden : DownloadedAssetDialogVisibilityState()
+    class Displayed (val assetName: String?, val assetData: ByteArray) : DownloadedAssetDialogVisibilityState()
+}
```

Comment:
i do not think we need to specify inside the ViewModel that we are giong to use a Dialog, we could have a sealead class like AssetDownloadStatus or something like that

Reply:
This is just an object that describes what is the state of the UI on the conversationViewState object, so it seems right to me. Maybe I'm not getting your point 🙈

Reply:
what I mean is that the ViewModel should not go in such details as managing the Dialog visibility in my opinion, ViewModel should be more agnostic of what the UI is going to do with the success/failure states, because perhaps later on we will not display the Dialog within this ViewModel, that should be up to View to decide, so instead of managing the DownloadedAssetDialogVisibilityState, you could rename it to  AssetDownloadStatus indicating the success state of the download